spyder/plugins/preferences/plugin.py

```diff
@@ -279,6 +282,23 @@ def on_toolbar_available(self):
             section=MainToolbarSections.ApplicationSection
         )
 
+    @on_plugin_available(plugin=Plugins.Application)
+    def on_application_available(self):
+        container = self.get_container()
+        container.sig_reset_preferences_requested.connect(self.reset)
+
+    @Slot()
+    def reset(self):
```

Comment:
I think we should move the code of this method to Application because it doesn't have to do with Preferences, but it's another kind of restart.

Then we could write this one (which is required due to the signal that comes from the container), like this

```python
@Slot()
def reset(self):
    application = self.get_plugin(Plugins.Application)
    application.reset()
```

Reply:
Respectfully, I disagree. Resetting the preferences is not really another restart, it is really just setting the environment variable. The current behavior includes restarting, so I didn't change that, but fundamentally this is a preferences plugin action. For example, in the abstract, hypothetically, if we removed the preferences plugin, there would be an orphan `reset` slot in the application plugin that would not mean anything.

Reply:
Ok, I think you're right. Let's leave it here then.